proxy/ControlMatcher.cc

```diff
@@ -791,71 +796,127 @@ ControlMatcher<Data, MatchResult>::BuildTableFromString(char *file_buf)
   int ip         = 0;
   int hostregex  = 0;
 
-  if (bufTok.Initialize(file_buf, SHARE_TOKS | ALLOW_EMPTY_TOKS) == 0) {
-    // We have an empty file
-    return 0;
-  }
-  // First get the number of entries
-  tmp = bufTok.iterFirst(&i_state);
-  while (tmp != nullptr) {
-    line_num++;
-
-    // skip all blank spaces at beginning of line
-    while (*tmp && isspace(*tmp)) {
-      tmp++;
+  if (!config_is_yaml) { // parse old line format.
+    if (bufTok.Initialize(file_buf, SHARE_TOKS | ALLOW_EMPTY_TOKS) == 0) {
+      // We have an empty file
+      return 0;
     }
+    // build a linked list of the entries
+    tmp = bufTok.iterFirst(&i_state);
+    while (tmp != nullptr) {
+      line_num++;
+
+      // skip all blank spaces at beginning of line
+      while (*tmp && isspace(*tmp)) {
+        tmp++;
+      }
 
-    if (*tmp != '#' && *tmp != '\0') {
-      const char *errptr;
-
-      current = (matcher_line *)ats_malloc(sizeof(matcher_line));
-      errptr  = parseConfigLine((char *)tmp, current, config_tags);
-
-      if (errptr != nullptr) {
-        if (config_tags != &socks_server_tags) {
-          Result error =
-            Result::failure("%s discarding %s entry at line %d : %s", matcher_name, config_file_path, line_num, errptr);
-          SignalError(error.message(), alarmAlready);
-        }
-        ats_free(current);
-      } else {
-        // Line parsed ok.  Figure out what the destination
-        //  type is and link it into our list
-        numEntries++;
-        current->line_num = line_num;
-
-        switch (current->type) {
-        case MATCH_HOST:
-        case MATCH_DOMAIN:
-          hostDomain++;
-          break;
-        case MATCH_IP:
-          ip++;
-          break;
-        case MATCH_REGEX:
-          regex++;
-          break;
-        case MATCH_URL:
-          url++;
-          break;
-        case MATCH_HOST_REGEX:
-          hostregex++;
-          break;
-        case MATCH_NONE:
-        default:
-          ink_assert(0);
+      if (*tmp != '#' && *tmp != '\0') {
+        current = (matcher_line *)ats_malloc(sizeof(matcher_line));
```

Comment:
I would use static_cast.  It's a good habit to use C++ casts.  With C-style casts, mistakes can cause tricky run-time bugs that would have been compile errors if you'd used a C++ cast.

Reply:
ok, that's all the existing code.  It was just re-formatted when i added in the yaml section.  I can change the casts.